.travis.yml

```diff
@@ -90,21 +90,73 @@ install:
   - export AWESOME_VERSION="${TRAVIS_BRANCH}-g$(git rev-parse --short HEAD)"
   - 'if [ "$TRAVIS_PULL_REQUEST" != false ]; then AWESOME_VERSION="${AWESOME_VERSION}-PR${TRAVIS_PULL_REQUEST}"; fi'
 
+  # Functions for custom Travis folds.
+  - |
+    travis_fold_start() {
+        echo -n "travis_"
+        echo "fold:start:$1"
+        travis_fold_current="$1"
+    }
+    travis_fold_end() {
+        echo -n "travis_"
+        echo "fold:end:$travis_fold_current"
+    }
```

Comment:
Perhaps it is nice to additionally do something like the following:
```sh
travis_run_in_fold() {
   travis_fold_start "$@"
   "$@"
   local result="$?"
   travis_fold_end "$@"
   ( exit "$result" )
}
```
Usage would be `travis_run_in_fold make`. Would this simplify some of the rest of the file.